From 0377fad75a2e74e570377968f92b5ed142750745 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 24 Feb 2020 18:49:09 +0800
Subject: [PATCH] video: rockchip: rga2: Fix compilation when !CONFIG_HIGHMEM

drivers/video/rockchip/rga2/rga2_mmu_info.c: In function 'rga2_dma_flush_page':
drivers/video/rockchip/rga2/rga2_mmu_info.c:69:11: error: implicit declaration of function 'kmap_high_get' [-Werror=implicit-function-declaration]

Change-Id: If389e8b1e61c1dcdffb94eb27d12e612a663193c
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/video/rockchip/rga2/rga2_mmu_info.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/rockchip/rga2/rga2_mmu_info.c b/drivers/video/rockchip/rga2/rga2_mmu_info.c
index 41ea3e55bdae..376cf291d225 100644
--- a/drivers/video/rockchip/rga2/rga2_mmu_info.c
+++ b/drivers/video/rockchip/rga2/rga2_mmu_info.c
@@ -60,6 +60,7 @@ static void rga2_dma_flush_page(struct page *page)
 	paddr = page_to_phys(page);
 #ifdef CONFIG_ARM
 	if (PageHighMem(page)) {
+#ifdef CONFIG_HIGHMEM
 		if (cache_is_vipt_nonaliasing()) {
 			virt = kmap_atomic(page);
 			dmac_flush_range(virt, virt + PAGE_SIZE);
@@ -69,6 +70,7 @@ static void rga2_dma_flush_page(struct page *page)
 			dmac_flush_range(virt, virt + PAGE_SIZE);
 			kunmap_high(page);
 		}
+#endif
 	} else {
 		virt = page_address(page);
 		dmac_flush_range(virt, virt + PAGE_SIZE);
-- 
2.35.3

